envi/codeflow.py

```diff
@@ -261,6 +261,9 @@ def addEntryPoint(self, va, arch=envi.ARCH_DEFAULT):
             cf.addEntryPoint( 0x77c70308 )
             ... callbacks flow along ...
         '''
+        # Architecture gets to decide on actual final VA and Architecture (ARM/THUMB/etc...)
+        va, arch = self._mem.arch.archModifyFuncAddr(va, arch)
+
```

Comment:
with above comment, this would probably be more like:
```
va, info = self._mem.arch.archModifyFuncAddr(va)
arch = info.get('arch',None) # (or whatever addCodeFlow is expecting to "stay in the arch we're in" i forget. )
```